test-suite/shared/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/logging/LoggerTest.scala

```diff
@@ -0,0 +1,362 @@
+      Logger.getLogger(s"$Prefix.TestLogger").getName)
+    // Level comes from the default log
+    assertNull(Logger.getLogger(s"$Prefix.TestLogger").getLevel)
+    assertTrue(Logger.getLogger(s"$Prefix.TestLogger").getUseParentHandlers)
+    assertNotNull(Logger.getLogger(s"$Prefix.TestLogger").getParent)
+  }
+
+  @Test def test_get_anonymous_logger(): Unit = {
+    assertEquals(null, Logger.getAnonymousLogger().getName)
+    // Level comes from the default log
+    assertNull(Logger.getAnonymousLogger().getLevel)
+    assertTrue(Logger.getAnonymousLogger().getUseParentHandlers)
+    assertNotNull(Logger.getAnonymousLogger().getParent)
+  }
+
+  @Test def test_properties(): Unit = {
+    val logger = Logger.getLogger(s"$Prefix.TestLogger2")
+    logger.setUseParentHandlers(true)
+    assertTrue(logger.getUseParentHandlers)
+    logger.setUseParentHandlers(false)
+    assertFalse(logger.getUseParentHandlers)
+
+    assertNull(logger.getResourceBundleName)
+
+    val filter = new TestFilter()
+    logger.setFilter(filter)
+    assertEquals(filter, logger.getFilter)
+
+    logger.setLevel(Level.WARNING)
+    assertEquals(Level.WARNING, logger.getLevel)
+  }
+
+  @Test def test_is_loggable(): Unit = {
+    val logger = Logger.getLogger(s"$Prefix.TestLogger3")
+    logger.setLevel(Level.ALL)
+    assertTrue(logger.isLoggable(logger.getLevel))
+
+    val levels = List(Level.SEVERE, Level.WARNING, Level.INFO, Level.CONFIG,
+      Level.FINE, Level.FINER, Level.FINEST)
+
+    assertTrue(levels.forall(logger.isLoggable))
+    logger.setLevel(Level.OFF)
+    assertFalse(levels.exists(logger.isLoggable))
+  }
+
+  @Test def test_is_loggable_inheritance(): Unit = {
+    val logger = Logger.getLogger(s"$Prefix.TestLogger4")
+    logger.setLevel(null)
+    logger.getParent.setLevel(Level.ALL)
+
+    val levels = List(Level.SEVERE, Level.WARNING, Level.INFO, Level.CONFIG,
+      Level.FINE, Level.FINER, Level.FINEST)
+
+    assertTrue(levels.forall(logger.isLoggable))
+    logger.getParent.setLevel(Level.OFF)
+    assertFalse(levels.exists(logger.isLoggable))
+  }
+
+  @Test def test_add_remove_handler(): Unit = {
+    val logger = Logger.getLogger(s"$Prefix.TestLogger5")
+    assertTrue(logger.getHandlers.isEmpty)
+    val testHandler = new TestHandler
+    logger.addHandler(testHandler)
+    assertTrue(logger.getHandlers.contains(testHandler))
+    logger.removeHandler(testHandler)
+    assertFalse(logger.getHandlers.contains(testHandler))
+  }
+
+  @Test def test_log_record_no_parents_all(): Unit = {
+    val logger = Logger.getLogger(s"$Prefix.TestLogger6")
+    val testHandler = new TestHandler
+    logger.setLevel(Level.ALL)
+    logger.addHandler(testHandler)
+    logger.setUseParentHandlers(false)
+
+    val lrFine = new LogRecord(Level.FINE, "Log")
+    logger.log(lrFine)
+    assertEquals(Some(Level.FINE), testHandler.lastRecord.map(_.getLevel))
+    assertEquals(Some("Log"), testHandler.lastRecord.map(_.getMessage))
```

Comment:
Consider changing `var lastRecord: LogRecord`, `assertEquals("Log", testHandler.lastRecord.getMessage)` and assume that in the assets they are non null. If they happens to be null then the test will fail with the NPE that will have more information about the source of the problem.